### ISSUE TITLE
[Feature Request] Allow for pre-configuring jsonServer.create() server

### ISSUE THREAD

**Issue by jimf** [typicode/json-server#63]
Hi! Thanks for JSON Server; it's awesome.

I've got a use case where I'm trying to run a json-server express instance, but I need to add some additional middleware *before* the static file handling gets added to the middleware stack. I began work toward enabling this, but was debating between a few varying solutions, so before I go any further I thought I'd ping you for your thoughts. Firstly, would you be on board with this enhancement? If so, I was thinking any of the following made sense:

 1. Update `create` to accept a callback. This callback would receive the `express()` server as an argument, which it could then use to do any configuration needed.
 1. Similar to `#1`, update `create` to accept an `options` object with a designated callback (e.g., `options.preconfigure`) to do the same configuration. This solution is arguably more future-proof.
 1. Update `create` to accept an optional express server instance, defaulting back to `express()` like it does currently. The downside to this is that it puts the onus of express dependency management on the user instead of in json-server.

Thoughts? I'd be happy to submit a PR for this.

**Comment by typicode** [comment]
Hi Jim, thanks for the feedback (and the talk btw :)).

Interesting issue and ideas. I tried a few things on my side (wasn't sure what was possible with Express) and ended with something like this:

```javascript
var server = jsonServer.create() // Barebone express() server
// You could add middlewares before here
server.use(jsonServer.defaults()) // Adds logger, public and CORS middlewares
server.use(jsonServer.router(db))
```

It adds a line but on the other side it has the advantage of being explicit and maybe more in the spirit of Express.

If that solves it, I'll commit and publish :)

**Comment by jimf** [comment]
+1. As long as you're cool with the [albeit small] breaking change, this sounds good to me. I agree that this feels more in line with express. Thanks!

**Comment by typicode** [comment]
Just released v0.7.0, should be good now :+1:

**Comment by jimf** [comment]
:+1: Thanks again!
